packages/compartment-mapper/src/compartment-map.js

```diff
@@ -13,6 +13,49 @@ const moduleLanguages = [
   'pre-cjs-json',
 ];
 
+/** @type {(a: string, b: string) => number} */
+// eslint-disable-next-line no-nested-ternary
+export const stringCompare = (a, b) => ((a === b ? 0 : a < b ? -1 : 1));
+
+/**
+ * @param {number} length
+ * @param {string} term
+ */
+const cumulativeLength = (length, term) => {
+  return length + term.length;
+};
+
+/**
+ * @param {Array<string>} a
+ * @param {Array<string>} b
+ */
+export const pathCompare = (a, b) => {
+  // Prefer the shortest dependency path.
+  if (a.length !== b.length) {
+    return a.length - b.length;
+  }
+  // Otherwise, favor the shortest cumulative length.
+  const aSum = a.reduce(cumulativeLength, 0);
```

Comment:
my lazy brain: why not just `a.join('').length`
It's obviously less performant but I wanted to know how much. 10 fold.